tests/rest/client/test_rooms.py

```diff
@@ -3163,6 +3181,27 @@ def test_threepid_invite_spamcheck(self) -> None:
             access_token=self.tok,
         )
         self.assertEqual(channel.code, 403)
+        self.assertEqual(channel.json_body.errcode, Codes.CONSENT_NOT_GIVEN)
+
+        # Also check that it stopped before calling _make_and_store_3pid_invite.
+        make_invite_mock.assert_called_once()
+
+        # Run variant with `Tuple[Codes, dict]`.
+        mock.return_value = make_awaitable((Codes.EXPIRED_ACCOUNT, {"field": "value"}))
+        channel = self.make_request(
+            method="POST",
+            path="/rooms/" + self.room_id + "/invite",
+            content={
+                "id_server": "example.com",
+                "id_access_token": "sometoken",
+                "medium": "email",
+                "address": email_to_invite,
+            },
+            access_token=self.tok,
+        )
+        self.assertEqual(channel.code, 403)
+        self.assertEqual(channel.json_body.errcode, Codes.EXPIRED_ACCOUNT)
+        self.assertEqual(channel.json_body.field, "value")
```

Comment:
Given we generally expect responses to be a JSON dict, let's treat it as such here:

```suggestion
        self.assertEqual(channel.json_body["errcode"], Codes.EXPIRED_ACCOUNT)
        self.assertEqual(channel.json_body["field"], "value")
```

(this is what is causing the CI to fail)